src/variable.js

```diff
@@ -36,11 +37,20 @@ Object.defineProperties(Variable.prototype, {
   _pending: {value: variable_pending, writable: true, configurable: true},
   _fulfilled: {value: variable_fulfilled, writable: true, configurable: true},
   _rejected: {value: variable_rejected, writable: true, configurable: true},
+  _resolve: {value: variable_resolve, writable: true, configurable: true},
   define: {value: variable_define, writable: true, configurable: true},
   delete: {value: variable_delete, writable: true, configurable: true},
   import: {value: variable_import, writable: true, configurable: true}
 });
 
+function initShadow(module, options) {
+  if (!options || !options.shadow) return null;
```

Comment:
Stricter types here? Allowing *options* to be false or zero would be surprising; that should error. I could see allowing *options*.shadow to be falsey though, I guess.

```suggestion
  if (!options?.shadow) return null;
```